ember-server/shared/src/main/scala/org/http4s/ember/server/EmberServerBuilder.scala

```diff
@@ -56,6 +56,7 @@ final class EmberServerBuilder[F[_]: Async] private (
     val additionalSocketOptions: List[SocketOption],
     private val logger: Logger[F],
     private val unixSocketConfig: Option[(UnixSockets[F], UnixSocketAddress, Boolean, Boolean)],
+    private val enableHttp2: Boolean,
```

Comment:
Is this always orthogonal to other settings?  If we opt into this, we speak 1 and 2?  Would there ever be a desire to speak 2 only?

Reply:
We could try to introduce http2 only support. I hadn't built that presently.

Reply:
I'm mostly concerned about the future state space there.  Now there are two states, so all is well:

* HTTP/1: `enableHttp2 = false`
* HTTP/1 and HTTP/2: `enableHttp2 = true`

If there's HTTP/2 only, now what?  You could introduce an `enableHttp1` flag, but what happens when they're both false?  Or you could preemptively make this a sum type, but you can't add to that without breaking bincompat.  Or you could make all three states now, but don't support the third state yet.  I don't think there's going to be a good answer here, but worst case, the server can fail to start on senseless arguments.